src/collectors/windows.plugin/perflib-mssql.c

```diff
@@ -730,17 +791,667 @@ static void do_mssql_errors(PERF_DATA_BLOCK *pDataBlock, struct mssql_instance *
+                                                              , PLUGIN_WINDOWS_NAME
+                                                              , "PerflibMSSQL"
+                                                              , PRIO_MSSQL_MEMMGR_PENDING_MEMORY_GRANTS
+                                                              , update_every
+                                                              , RRDSET_TYPE_LINE
+                                                              );
+
+            snprintfz(id, RRD_ID_LENGTH_MAX, "mssql_instance_%s_memmgr_pending_memory_grants", p->instanceID);
+            p->rd_pending_mem_grant  = rrddim_add(p->st_pending_mem_grant,
+                                                 id,
+                                                 "pending",
+                                                 1,
+                                                 1,
+                                                 RRD_ALGORITHM_INCREMENTAL);
+
+            rrdlabels_add(p->st_pending_mem_grant->rrdlabels, "mssql_instance", p->instanceID, RRDLABEL_SRC_AUTO);
+        }
+
+        rrddim_set_by_pointer(p->st_pending_mem_grant,
+                              p->rd_pending_mem_grant,
+                              (collected_number)p->MSSQLPendingMemoryGrants.current.Data);
+        rrdset_done(p->st_pending_mem_grant);
+    }
+
+    if (perflibGetObjectCounter(pDataBlock, pObjectType, &p->MSSQLPendingMemoryGrants)) {
+        snprintfz(id, RRD_ID_LENGTH_MAX, "instance_%s_memmgr_server_memory", p->instanceID);
+        if (!p->st_mem_tot_server) {
+            p->st_mem_tot_server = rrdset_create_localhost("mssql"
+                                                           , id, NULL
+                                                           , "memory"
+                                                           , "mssql.instance_memmgr_server_memory"
+                                                           , "Memory committed"
+                                                           , "bytes"
+                                                           , PLUGIN_WINDOWS_NAME
+                                                           , "PerflibMSSQL"
+                                                           , PRIO_MSSQL_MEMMGR_TOTAL_SERVER
+                                                           , update_every
+                                                           , RRDSET_TYPE_LINE
+                                                           );
+
+            snprintfz(id, RRD_ID_LENGTH_MAX, "mssql_instance_%s_memmgr_total_server_memory_bytes", p->instanceID);
+            p->rd_mem_tot_server  = rrddim_add(p->st_mem_tot_server,
+                                              id,
+                                              "memory",
+                                              1,
+                                              1,
+                                              RRD_ALGORITHM_INCREMENTAL);
```

Comment:
Is this incremental? Please check the others as well

Reply:
I also checked others and I push a commit to address them. Thanks!